src/utils/oauth2_util.py

```diff
@@ -39,7 +67,7 @@ def build_user_credential_access_token(user, client, secret_key, scopes):
             "username": user.username,
             "name": user.name,
             "last_name": user.last_name,
-            "roles": user.roles,
+            "roles": [user.roles],
             "scopes": scopes,
```

Comment:
I think you need to do a split here, We do know that you can have a least to scopes, `read and write`. They can be comma separated

Reply:
done.